cmake/helpers.cmake

```diff
@@ -104,3 +108,31 @@ function(add_umf_library)
     add_umf_target_compile_options(${ARG_NAME})
     add_umf_target_link_options(${ARG_NAME})
 endfunction()
+
+# Add sanitizer ${flag}, if it is supported, for both C and C++ compiler
+macro(add_sanitizer_flag flag)
+	set(SAVED_CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES})
+	set(CMAKE_REQUIRED_LIBRARIES "${CMAKE_REQUIRED_LIBRARIES} -fsanitize=${flag}")
+
+	if(${flag} STREQUAL "address")
+		set(check_name "HAS_ASAN")
+	elseif(${flag} STREQUAL "undefined")
+		set(check_name "HAS_UBSAN")
+	endif()
+
+	check_c_compiler_flag("-fsanitize=${flag}" "C_${check_name}")
+	check_cxx_compiler_flag("-fsanitize=${flag}" "CXX_${check_name}")
+	if (${C_${check_name}} OR ${CXX_${check_name}})
+		set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=${flag} -fno-sanitize-recover=all")
+		if (${C_${check_name}})
+			set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fsanitize=${flag} -fno-sanitize-recover=all")
+		endif()
+		if (${CXX_${check_name}})
+			set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=${flag} -fno-sanitize-recover=all")
+		endif()
+	else()
+		message(STATUS "  ${flag} sanitizer is not supported (neither by C nor CXX compiler)")
```

Comment:
We should fail if we cannot configure the project in the way the user wants it

Reply:
Done.